### ISSUE TITLE
Cannot focus on an input that is inside a MenuItem

### ISSUE THREAD

**Issue by rogor** [chakra-ui/chakra-ui#2111]
# Bug report

## Describe the bug

Cannot focus on an input that is inside a MenuItem. When I click I immediately lose focus and cannot use the input field.
I have tried both a simple html <input> tag and the different chakra input components with the same result.

## To reproduce

Code: 

```
<Menu closeOnSelect={false}>
  <MenuButton
    as={IconButton}
    aria-label="settings"
    isRound
    icon={<RiSettings4Fill />}
    variant="ghost"
  />
  <MenuList bg={menuBg}>
    <MenuGroup title="Add Delay">
      <MenuItem>
        <input id="delay" name="delay" />
      </MenuItem>
    </MenuGroup>
  </MenuList>
</Menu>
```

Result: 
![2020-09-28 16-37-00](https://user-images.githubusercontent.com/2693832/94448165-e308b980-01aa-11eb-8ca2-6b9673a0aaca.gif)

## Expected behavior

The input stays in focus after I click on it.

## System information

- OS: Windows 10
- Browser (if applies):  Any
- Version of @chakra-ui/core: 1.0.0-rc.5
- Version of Node.js: v14.8.0

**Comment by rogor** [comment]
After some testing with Chrome dev tools, it seems that the focus immediately goes back to the MenuList element. 
The MenuList doesn't seem to have any event listener though, so I don't understand how it steals the focus when I click.

**Comment by Psvensso** [comment]
Was having issues with a similar thing. 
Is it trying to close the menu? I think it's "interacting outside" and therefore closing the menu.
But a break inside the "onclose focus back" effect [here](https://github.com/chakra-ui/chakra-ui/blob/e1f2b61115ee60b227f3df39b3372e728268a3ca/packages/menu/src/use-menu.ts#L162)

**Comment by rogor** [comment]
Hmm, from my testing this PR #2118 doesn't fix the problem. I don't think the two problems are related.

**Comment by ljosberinn** [comment]
Oops, typo'd the issue, sorry!

**Comment by rogor** [comment]
@Psvensso, it doesn't seem to be trying to close the menu, no.
From what I can tell, the only piece of code that runs from this file when I click is [the click event handler](https://github.com/chakra-ui/chakra-ui/blob/develop/packages/menu/src/use-menu.ts#L445), so nothing seems wrong here. 
I spent a few hours on this and I can't figure it out.

I've tried using react-focus-lock to see what would happen:
```
<FocusLock>
  <NumberInput
    defaultValue={delay}
    onChange={setDelay}
    >
      <NumberInputField />
  </NumberInput>
</FocusLock>
```
With this, I do get the focus inside the input (with the cursor), but I can't actually type anything in it.

After a bunch of googling, I've seen people having the same problem with other menu libraries, it seems like putting inputs inside menus isn't usually supported. I'll just have to create my own menu component, it shouldn't be too bad :p

**Comment by ljosberinn** [comment]
@rogor I've been silently following the convo because I didn't have time yet to investigate this - would you mind linking resources where this is an issue in other libs too?

**Comment by rogor** [comment]
@ljosberinn I've seen a similar problem here with react-bootstrap: https://github.com/react-bootstrap/react-bootstrap/issues/62

**Comment by ljosberinn** [comment]
it's been too silent here - we're aware of the issue and will revisit it once #2200 lands

**Comment by Samic8** [comment]
I tried debugging this by commenting out any manual focus that we do in the menu hooks and components. Still no luck, the input still was not maintaining focus on click.

**Comment by jessicalc** [comment]
Thanks for filing this!

I'm not sure if this is related, but I think we ran into something similar. In our implementation, we opened a `Modal` (which includes an `Input` field) on clicking on a `MenuItem`, but the input field is unresponsive. We can't fire an `change`/`keypress` events on typing stuff into it, but we can fire `click` events. 

If we make it so that `URLMenuItem` component is not nested inside `MenuList`, it works fine... 

```
export const URLMenuItem = () => {
  const { isOpen, onOpen, onClose } = useDisclosure()
  const [url, setUrl] = React.useState('')
  const onSubmit = () => {
    // implement later
  }
  return (
    <>
      <MenuItem onClick={onOpen}>URL</MenuItem>
      <Modal isOpen={isOpen} onClose={onClose} size="md">
        <ModalOverlay />
        <ModalContent>
          <ModalHeader>Choose a URL</ModalHeader>
          <ModalCloseButton />
          <ModalBody>
            <VStack align="flex-start">
              <Input
                onChange={(e) => setUrl(e.target.value)}
                placeholder="Paste in a URL here"
                value={url}
              />
            </VStack>
          </ModalBody>
          <ModalFooter>
            <Button colorScheme="blue" onClick={onSubmit}>
              Go to URL
            </Button>
          </ModalFooter>
        </ModalContent>
      </Modal>
    </>
  )
}
```

**Comment by xJkit** [comment]
Any update here?

 It seems that all input component inside MenuList component is not working

**Comment by tomdohnal** [comment]
I wonder if placing inputs inside the `Menu` component should even be supported as I don't believe that's the use-case for a `role="menu"` component as per https://www.w3.org/TR/wai-aria-1.1/#menu

> menu (role)§
>
>A type of widget that offers a list of choices to the user.
>
>A menu is often a list of common actions or functions that the user can invoke. The menu role is appropriate when a list of >menu items is presented in a manner similar to a menu on a desktop application.

I believe that if you put an `input` into a `Menu` then it's not a menu in terms of being a "type of widget that offers a list of choices to the user."

**Comment by WolfyUK** [comment]
We also experienced a similar issue whereby a `<MenuItem>` opened a `<Modal>` with `<Input>`s inside. Focus was not in this case, but the `<Input>`s seemed to struggle to manage their state resulting in a 'read only' effect. After finding this issue we refactored to move the `<Modal>` from outside the `<Menu>` and now all is well. 

I originally thought this was a `<Formik>` issue, but just moving a standalone `<Input>` inside/outside the `<Menu>` proved the issue, so thanks for logging this!
